### ISSUE TITLE
Storing wellplate state

### ISSUE THREAD

**Issue by braceal** [braceal/ot2util#1]
It might be helpful to maintain a data structure of the available wells in each well plate so that we don't have to manually specify which destination wells to use during a search strategy.

This could be abstracted with a WellPlate data structure.

The search agent could run until all the wells are full and then pause for human input. Someone could clear the well plates, add more, and then communicate this to the agent via an input handler.

**Comment by KPHippe** [comment]
This would also be enhanced by some type of self-monitoring system and cell manipulation tool like we talked about the other day. 

As a side note, is there a way to stop the protocol at some point? Say we reached the goal, it seems wasteful to continue the search, so is there some way of stopping a protocol mid execution?

**Comment by braceal** [comment]
Yes, that can be done. The way it's currently setup in the grid search example, each "experiment" requires a unique call of opentrons_execute. So the search strategy generates new "experiments" as it goes and runs each sequentially, meaning that it could be stopped at any time by some convergence criteria logic (or a simple ^C of the search program). Also, it might not be difficult for a human to interact with the search strategy by suggesting experiments via config files mid way through a search.
